src/main/java/appeng/me/cache/NetworkMonitor.java

```diff
@@ -121,15 +285,18 @@ protected void postChange( final boolean add, final Iterable<T> changes, final B
 
 			if( !add && changedItem != null )
 			{
-				( difference = changedItem.copy() ).setStackSize( -changedItem.getStackSize() );
+				difference = changedItem.copy();
+				difference.setStackSize( -changedItem.getStackSize() );
 			}
 
 			if( this.myGridCache.getInterestManager().containsKey( changedItem ) )
 			{
 				final Collection<ItemWatcher> list = this.myGridCache.getInterestManager().get( changedItem );
+
 				if( !list.isEmpty() )
 				{
-					IAEStack fullStack = myStorageList.findPrecise( changedItem );
+					IAEStack fullStack = this.getStorageList().findPrecise( changedItem );
+
 					if( fullStack == null )
 					{
 						fullStack = changedItem.copy();
```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) NullPointerException might be thrown as 'changedItem' is nullable here [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.orod.de/coding_rules#rule_key=squid%3AS2259)